code/modules/power/engines/tesla/energy_ball.dm

```diff
@@ -33,6 +34,18 @@
 	var/energy_to_lower = -20
 	var/list/shocked_things = list()
 	var/obj/singularity/energy_ball/parent_energy_ball
+	/// Variable to determine if it has a target
+	var/has_a_target = FALSE
+	/// All the turfs in the chosen area
+	var/list/target_area_turfs = list()
+	/// Turf where the tesla will move to if it's loose
+	var/turf/target_turf
+	/// List with the turfs in the line towards the target turf
+	var/list/tesla_line = list()
```

Comment:
why use this line method when you could use `get_step` and `get_dir` each time? the line worries me holding refs to a bunch of turfs especially when a tesla is loose.

Reply:
Mostly because someone recommended it to me as being the exact thing that I was looking for for this idea, and because it's not used often.
Is there anything I can do to make it more secure, or is moving over to get_step and get_dir a better idea?

Reply:
Hal is right, you might need to move over. If say a wall is in this list and the tesla zaps it, we're gonna have GC issues. @GDNgit might be able to give some insight into that.